master/buildbot/configurators/janitor.py

```diff
@@ -39,43 +40,67 @@ class LogChunksJanitor(BuildStep):
     name = 'LogChunksJanitor'
     renderables = ["logHorizon"]
 
-    def __init__(self, logHorizon):
+    def __init__(self, logHorizon=None, horizonPerBuilder=None):
         super().__init__()
         self.logHorizon = logHorizon
+        self.horizonPerBuilder = horizonPerBuilder
 
     @defer.inlineCallbacks
     def run(self):
-        older_than_timestamp = datetime2epoch(now() - self.logHorizon)
-        deleted = yield self.master.db.logs.deleteOldLogChunks(older_than_timestamp)
-        self.descriptionDone = ["deleted", str(deleted), "logchunks"]
+        if self.logHorizon is not None:
+            older_than_timestamp = datetime2epoch(
+                now() - self.logHorizon)
+            deleted = yield self.master.db.logs.deleteOldLogChunks(
+                older_than_timestamp=older_than_timestamp)
+            self.descriptionDone = ["deleted", str(deleted), "logchunks"]
+
+        if self.horizonPerBuilder is not None:
+            deleted = yield self.master.db.logs.deleteOldLogChunks(
+                horizonPerBuilder=self.horizonPerBuilder)
+            self.descriptionDone = ["deleted", str(deleted), "logchunks"]
+
         return SUCCESS
 
 
 class BuildDataJanitor(BuildStep):
     name = 'BuildDataJanitor'
     renderables = ["build_data_horizon"]
 
-    def __init__(self, build_data_horizon):
+    def __init__(self, build_data_horizon=None, horizonPerBuilder=None):
         super().__init__()
         self.build_data_horizon = build_data_horizon
+        self.horizonPerBuilder = horizonPerBuilder
 
     @defer.inlineCallbacks
     def run(self):
-        older_than_timestamp = datetime2epoch(now() - self.build_data_horizon)
-        deleted = yield self.master.db.build_data.deleteOldBuildData(older_than_timestamp)
-        self.descriptionDone = ["deleted", str(deleted), "build data key-value pairs"]
+        if self.build_data_horizon is not None:
+            older_than_timestamp = datetime2epoch(now() - self.build_data_horizon)
+            deleted = yield self.master.db.build_data.deleteOldBuildData(
+                older_than_timestamp=older_than_timestamp)
+            self.descriptionDone = ["deleted", str(deleted), "build data key-value pairs"]
+
+        if self.horizonPerBuilder is not None:
+            deleted = yield self.master.db.build_data.deleteOldBuildData(
+                horizonPerBuilder=self.horizonPerBuilder)
+            self.descriptionDone = ["deleted", str(deleted), "build data key-value pairs"]
         return SUCCESS
 
 
 class JanitorConfigurator(ConfiguratorBase):
     """ Janitor is a configurator which create a Janitor Builder with all needed Janitor steps"""
 
-    def __init__(self, logHorizon=None, hour=0, build_data_horizon=None, **kwargs):
+    def __init__(self, logHorizon=None, hour=0, build_data_horizon=None, horizonPerBuilder=None,
+    **kwargs):
```

Comment:
weird indentation

Reply:
```
    def __init__(self, logHorizon=None, hour=0, build_data_horizon=None, horizonPerBuilder=None,
        **kwargs):
```
leads to flake8 error E125.
<br />
```
    def __init__(self,
                 logHorizon=None,
                 hour=0,
                 build_data_horizon=None,
                 horizon_per_builder=None,
                 **kwargs):
```
should it be like this?